packages/graphql/src/schema-model/entity/ConcreteEntity.ts

```diff
@@ -71,6 +71,17 @@ export class ConcreteEntity implements Entity {
         this.attributes.set(attribute.name, attribute);
     }
 
+    private deleteAttribute(attributeName: string): void {
+        if (this.attributes.has(attributeName)) {
+            this.attributes.delete(attributeName);
+        }
+    }
+
+    public attributeToRelationship(relationship: Relationship): void {
+        this.addRelationship(relationship);
+        this.deleteAttribute(relationship.name);
+    }
```

Comment:
I guess that this is due to a workflow a bit like:

* Add all attributes
* Remove those that are relationship

Is there a way of doing these checks outside the schema model? Ideally, all of this logic exists in the factories, so the schema model is somehow static

Reply:
so you mean exposing deleteAttribute and doing this reassignment in the generate-model?

Reply:
Mmm, I'm thinking more of figuring out which are relationships and which are attributes before setting them into the schema model.

I'm just saying that because for me it is easier to reason about a stateless model than to modify it after creating it. Just as an alternative to consider :sweat_smile:

Reply:
ah yes I see what you mean - I agree with you that ideally this should be the case. However, that requires changing the way the schema model is being generated, which I plan to do but I wanted to have interfaces and unions in first, in order to unblock the big refactors as quickly as possible. I will assess this!

Reply:
you can revisit this - I changed it from an after-the-fact "hydrate" method that mutates the class to during construction.